sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java

```diff
@@ -0,0 +1,265 @@
+            } else if ("POST".equalsIgnoreCase(httpMethod) && pollingContext.getData(LOCATION) != null) {
+                finalGetUrl = pollingContext.getData(LOCATION);
+            } else {
+                throw logger.logExceptionAsError(new RuntimeException("Cannot get final result"));
+            }
+        }
+
+        if (finalGetUrl == null) {
+            String latestResponseBody = pollingContext.getData(POLL_RESPONSE_BODY);
+            if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, resultType.getJavaType())) {
+                return (Mono<U>) Mono.just(BinaryData.fromString(latestResponseBody));
+            } else {
+                return Mono.fromCallable(() -> serializer.deserialize(latestResponseBody, resultType.getJavaType(),
+                    SerializerEncoding.JSON));
+            }
+        } else {
+            HttpRequest request = new HttpRequest(HttpMethod.GET, finalGetUrl);
+            Mono<HttpResponse> responseMono;
+            if (context == null) {
+                responseMono = httpPipeline.send(request);
+            } else {
+                responseMono = httpPipeline.send(request, context);
+            }
+            return responseMono.flatMap(res -> {
+                if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, resultType.getJavaType())) {
+                    return (Mono<U>) BinaryData.fromFlux(res.getBody());
+                } else {
+                    return res.getBodyAsString().flatMap(body -> Mono.fromCallable(() ->
+                        serializer.deserialize(body, resultType.getJavaType(), SerializerEncoding.JSON)));
+                }
+            });
+        }
+    }
+
+    @Override
+    public Mono<T> cancel(PollingContext<T> pollingContext, PollResponse<T> initialResponse) {
+        return Mono.error(new IllegalStateException("Cancellation is not supported."));
+    }
+
+    /**
+     * A simple structure representing the partial response received from an operation location URL, containing the
+     * information of the status of the long running operation.
+     */
+    private static class PollResult {
+        private LongRunningOperationStatus status;
+        private String resourceLocation;
+
+        /**
+         * Gets the status of the long running operation.
+         * @return the status represented as a {@link LongRunningOperationStatus}
+         */
+        public LongRunningOperationStatus getStatus() {
+            return status;
+        }
+
+        /**
+         * Sets the long running operation status in the format of a string returned by the service. This is called by
+         * the deserializer when a response is received.
+         *
+         * @param status the status of the long running operation
+         * @return the modified PollResult instance
+         */
+        @JsonSetter
+        public PollResult setStatus(String status) {
+            if ("NotStarted".equalsIgnoreCase(status)) {
```

Comment:
Maybe you can create an enum class or make it a constant.

Reply:
Moving to a static final field for now. We can move to an enum or `Constants` class if we need it again in the future

Reply:
Moved to `implementation/PollingConstants.java`